### ISSUE TITLE
split-apply-combine

### ISSUE THREAD

**Issue by antagomir** [microbiome/miaTime#30]
Many real applications require calculating time series analyses per subject. In fact this is more generic problem that can be solved on the same go for any grouping (not just subject, and not just time series).

Two cases:
1. For the full TreeSE object
2. For colData only (adding back to TreeSE; can be faster, if only colData is manipulated)

This is needed in many other functions.

Design and implement a general solution that takes as input 
- the SE object
- the grouping variable
- name of the function to apply
.. and then returns a new SE object where that function has been applied per group and combined back into a single SE object

Perhaps two functions: one for each case described above.

I am tagging @TuomasBorman here, he might find this interesting as well and could join the discussions. @YagmurSimsekk could give it the first try.

**Comment by antagomir** [comment]
Functions `mia::splitOn` and `mia::unsplitOn` might help here now. This may require some more thinking.

**Comment by antagomir** [comment]
This function could go to mia as well as it is so fundamental, not sure which package would be best. This doesnt necessarily add dependencies.

@TuomasBorman what's your view: mia or miaTime?

**Comment by TuomasBorman** [comment]
Splitting TreeSE into a list of TreeSEs is not memory efficient. Instead, I think we should take this grouping into account internally in the methods that we develop (`group` parameter). Then we can use `dplyr` or loop through the subjects, but it is still faster because we are not copying the whole structure of TreeSE

There might be room for split-apply-combine (mia would be the right place if we decide to implement this as this is more general tool), but in practice this would only remove couple lines of code (and potentially make the functions harder to understand).

```
tse_list <- splitOn(tse, by = 2, group = "subject")
tse_list <- lapply(tse_list, function(x) doSomething(x, param1 = TRUE, param2 = FALSE))
tse <- mergeSEs(tse_list)
```
vs

`tse <- splitCombine(tse, by = 2, group = "subject", FUN = doSomething, param1 = TRUE, param2 = FALSE)`

**Comment by antagomir** [comment]
True. Let us now implement such functionality now but it would be very useful to have an example in miaTime vignette. 

There is section "Operations by unit" already in the [miaTime vignette](https://microbiome.github.io/miaTime/articles/manipulation.html). We could add the above procedure there for a future reference.  @Daenarys8 could you suggest/add an example?

Or we could add it in OMA but miaTime is not in Bioconductor yet, and might be generally easier to first accumulate material in the package vignette, then later see if there is space for an extra chapter in OMA and do that properly at one go.

**Comment by TuomasBorman** [comment]
This issue was discussed in mia (which would have been the correct place for this kind of general function): https://github.com/microbiome/mia/issues/659
